src/reconcile/rec_write.c

```diff
@@ -1798,7 +1808,7 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
 
     /* Check if there are saved updates that might belong to this block. */
     if (r->supd_next != 0)
-        WT_RET(__rec_split_write_supd(session, r, chunk, multi, last_block));
+        WT_RET(__rec_split_write_supd(session, r, chunk, multi, last_block, &supd_restore));
 
```

Comment:
I would like clarification on one thing here, in __rec_supd_move we get the supd and apply it to the multi and keep a pointer to whether that supd needed restoration. We then apply that pointer in __rec_split_write to the same multi as far as I can tell. Why can't we apply it to the multiple when we move it onto the multi? Is it because of the checks you do in __rec_split_write? i.e. WT_REC_IN_MEMORY or WT_REC_EVICT flag checks.

Reply:
Good point. I simplified it and removed the pointer.